### PR TITLE
wordsmithing the multiple inheritance chapter

### PR DESCRIPTION
**Pull request by sichvoge**
according to #407

### CODE REVIEW

**Comment by krishahn** [comment]
Editorial suggestions committed.

**Comment by usarid** [comment]
Now that we've removed the restriction on multiple inheritance being only for object types, we must specify whether you can have multiple inheritance from types that aren't extensions of the same built-in types, e.g. `type: [ number, string ]` or `type: [ number, integer ]` or `type: [ object, string ]` or `type: [ (string | boolean), string ]` or `type: [ foo, bar ]` where `foo` and `bar` are user-defined types of various sorts. @petrochenko-pavel-a can help clarify?

**Comment by petrochenko-pavel-a** [comment]
>whether you can have multiple inheritance from types that aren't extensions of the same built-in types
There is no need to specially restrict it. But this types will be invalid, because they will have contra-dictional restrictions on they instanced for example `[number, boolean]` will have a restriction that instance should be javascript number and boolean at same moment

**Comment by sichvoge** [comment]
Is there any general rule we need to apply or mention? @petrochenko-pavel-a w

**Comment by petrochenko-pavel-a** [comment]
The generic rules is that type is invalid if it has any conflicting restrictions.

**Comment by sichvoge** [comment]
> conflicting restrictions
maybe a very dumb question, but what does that mean exactly?

**Comment by sichvoge** [comment]
OK so referring to that I'd suggest to call out the restrictions that you are not allowed to have different primitives participating in multiple inheritance.

**Comment by sichvoge** [comment]
See last commit.

File: versions/raml-10/raml-10.md [hunk 1]
```diff
@@ -1227,26 +1227,53 @@ When referencing an inner element of a schema, a RAML processor MUST validate an
...
+    properties:
+      employeeNr: integer
+  Teacher:
+    type: [ Person, Employee ]
 ```
 
-Note: Multiple inheritance is only allowed if all Type Expressions are simple object Types.
+In the example above, the type `Teacher` inherits all restrictions from `Person` and `Employee`.
+
+Multiple inheritance is allowed only if the sub-type is still a valid type declaration after inheriting all restrictions from its parent types. Also, it is not allowed to inherit from different kind of primitive types, for example `[ number, string ]`.
```
Comment (sichvoge) [inline versions/raml-10/raml-10.md]: Uri's comment:

> I think you also need to explicitly state how to treat this statement when you have union types somewhere in the inheritance chain. I would guess the answer is to expand all the unions and and see that at least one of the allowed combinations is a valid type declaration and in particular does not combine different primitive types, but @petrochenko-pavel-a should say definitively.

Comment (usarid) [inline versions/raml-10/raml-10.md]: I'm fine as long as it appears somewhere in the spec; in that case, you might want to just link to it from here.

**Pull request merged**